src/network/mcpe/NetworkSession.php

```diff
@@ -1170,6 +1171,10 @@ public function onFormSent(int $id, Form $form) : bool{
 		return $this->sendDataPacket(ModalFormRequestPacket::create($id, json_encode($form, JSON_THROW_ON_ERROR)));
 	}
 
+	public function onCloseAllForms() : void{
+		$this->sendDataPacket(ClientboundCloseFormPacket::create());
```

Comment:
This function is unnecessary, it will never be sent by the client. The reason it has a handle is for proxy purposes

Reply:
You have misunderstood the function of the method, `NetworkSession::onCloseAllForms()` is a wrapper to send `ClientboundCloseFormPacket`, and is called by `Player::closeAllForms()`.

Reply:
This allows not mixing the API with the protocol and allows multi-version implementations to handle the packet as they wish.